src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -155,8 +160,20 @@ private MessageConsumer newInitialStream(
         long connectAuthorization = acceptAuthorization;
         if (verified != null)
         {
-            final String kid = verified.getKeyIdHeaderValue();
-            connectAuthorization = resolveRealm.applyAsLong(kid);
+            // Try to parse the scopes claim if it exists.
+            try
+            {
+                final String kid = verified.getKeyIdHeaderValue();
+                final JwtClaims claims = JwtClaims.parse(signature.getPayload());
+                final Object scopeClaim = claims.getClaimValue(SCOPES_CLAIM);
+                connectAuthorization = scopeClaim != null ?
+                                resolveRealm.applyAsLong(kid, splitScopes(scopeClaim.toString()))
+                                : resolveRealm.applyAsLong(kid, null);
+            }
+            catch (JoseException | InvalidJwtException ex)
+            {
+                // TODO: diagnostics?
+            }
```

Comment:
Consider changing `resolveRealm` to `ToLongFunction<JsonWebSignature>` instead, so that all the exception handling and knowledge of specific `scopes` claim mapping to our security roles is self-contained inside `OAuthRealms`.